pkg/sanity/volume.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2017 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sanity
+
+import (
+	"strconv"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/api/client"
+	volumeclient "github.com/libopenstorage/openstorage/api/client/volume"
+
+	"github.com/libopenstorage/openstorage/volume"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Volume [Volume Tests]", func() {
+	var (
+		restClient   *client.Client
+		volumedriver volume.VolumeDriver
+	)
+
+	BeforeEach(func() {
+		var err error
+		restClient, err = volumeclient.NewDriverClient(osdAddress, "nfs", volume.APIVersion, "")
+
+		Expect(err).ToNot(HaveOccurred())
+		//volumedriver = clusterclient.ClusterManager(restClient)
+		volumedriver = volumeclient.VolumeDriver(restClient)
+	})
+
+	AfterEach(func() {
+
+	})
+
+	Describe("Volume Create", func() {
+
+		var (
+			volumerequest *api.VolumeCreateRequest
+			volumeID      string
+		)
+
+		BeforeEach(func() {
+
+			var err error
+
+			volumerequest = &api.VolumeCreateRequest{
+				Locator: &api.VolumeLocator{
+					Name: "vol-osd-sanity-cd",
+					VolumeLabels: map[string]string{
+						"class": "sanity-test-class",
+					},
+				},
+				Source: &api.Source{},
+				Spec: &api.VolumeSpec{
+					Size:   1,
+					Format: api.FSType_FS_TYPE_NFS,
+				},
+			}
+
+			volumeID, err = volumedriver.Create(volumerequest.GetLocator(), volumerequest.GetSource(), volumerequest.GetSpec())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(volumeID).ToNot(BeNil())
+		})
+
+		AfterEach(func() {
+			err := volumedriver.Delete(volumeID)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("Volume should be present in inspect", func() {
+
+			inspectVolumes := []string{volumeID}
+
+			By("only one volume to be returned and having the same name as the created volume in beforeeach.")
```

Comment:
For this to be true, the test should confirm that there are no volumes already created, which could be complicated. This is why creating the volume in the It() would be easier, specially for this algorithm:

* Save the number of volumes already in the cluster.
* Create volume
* Expect that now there is 1 more volume
* Get information on volume
* Delete volume

Reply:
You can also save the current number of volumes in the cluster on BeforeEach before calling volume create.